pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -507,9 +518,41 @@ async def client_streams(
             MemoryObjectSendStream[SessionMessage],
         ]
     ]:
+        r_fd, w_fd = os.pipe()
+        err_read_stream_writer, err_read_stream = anyio.create_memory_object_stream[str]()
+        stderr_chunks = []
+
         server = StdioServerParameters(command=self.command, args=list(self.args), env=self.env, cwd=self.cwd)
-        async with stdio_client(server=server) as (read_stream, write_stream):
-            yield read_stream, write_stream
+
+        async def write_stderr():
+            async with err_read_stream_writer:
+                async with await anyio.open_file(r_fd, encoding=server.encoding) as stream:
+                    line = await stream.read()
+                    if line:
+                        await err_read_stream_writer.send(line)
+
+        async def read_stderr():
+            async with err_read_stream:
+                async for chunk in err_read_stream:
+                    stderr_chunks.append(chunk)
+
+        raise_stderr_exception = False
+        async with anyio.create_task_group() as tg:
+            tg.start_soon(write_stderr)
+            tg.start_soon(read_stderr)
+            try:
+                async with stdio_client(server=server, errlog=w_fd) as (read_stream, write_stream):
+                    os.close(w_fd)
+                    yield read_stream, write_stream
+            except ExceptionGroup as eg:
+                mcp_errors = [error for error in iter_exc(eg) if isinstance(error, McpError)]
+                if mcp_errors and mcp_errors[0].error.message == 'Connection closed':
+                    raise_stderr_exception = True
```

Comment:
Instead of storing a boolean, let's store the exception itself so we can reference it from the bit below instead of `mcp_errors[0]`